k8s/upgrade/src/upgrade_resources/upgrade.rs

```diff
@@ -137,13 +137,77 @@ pub struct GetUpgradeArgs {}
 impl GetUpgradeArgs {
     ///  Upgrade the resources.
     pub async fn get_upgrade(&self, namespace: &str) {
-        println!("{namespace}")
-        // To be implemented
+        // Create resources for getting upgrade status
+        match GetUpgradeEventClient::create_get_upgrade_resource(namespace).await {
+            Ok(_) => {}
+            Err(error) => eprintln! {"Error in getting upgrade status: {error}"},
+        }
+    }
+}
+
+/// Resource to be created to get upgrade status.
+struct GetUpgradeEventClient {
+    upgrade_event: Api<Event>,
+}
+
+/// Methods implemented by GetUpgradeEventClient.
+impl GetUpgradeEventClient {
+    pub async fn new(ns: &str) -> Result<Self, Error> {
+        let client = Client::try_default().await?;
+        Ok(Self {
+            upgrade_event: Api::<Event>::namespaced(client, ns),
+        })
+    }
+
+    /// getter method to return upgrade event
+    pub(crate) fn get_upgrade_event(&self) -> Api<Event> {
```

Comment:
Please make the doc comments start with capital and end with `.`. Please fix these below as well at all places.